File: boj8/2903.cc
// 중앙 이동 알고리즘
// https://www.acmicpc.net/problem/2903

#include <stdio.h>

int N;

int main() {
  scanf("%d", &N);
  int base = 2;
  for (int i = 0; i < N; i++) {
    base = base * 2 - 1;
  }
  printf("%d\n", base * base);
  return 0;
}

File: past_archive/23boj_py/2903.py
# 중앙 이동 알고리즘
# https://www.acmicpc.net/problem/2903

N = int(input())

num_pts = 4
s = 1
while N > 0:
    num_sqr = s ** 2
    num_pts += num_sqr * 5 - 2 * s * (s - 1)
    s *= 2
    N -= 1

print(num_pts)
